x-pack/solutions/search/plugins/search_playground/public/components/query_mode/query_viewer.tsx

```diff
@@ -76,10 +72,14 @@ export const ElasticsearchQueryViewer = ({
         setEsQueryFirtEdit(true);
         usageTracker?.count(AnalyticsEvents.editElasticsearchQuery);
       }
-      onChangeUserQuery(value);
+      onChangeUserQuery(value === generatedEsQuery ? null : value);
+      trigger(PlaygroundFormFields.userElasticsearchQuery);
     },
-    [esQueryFirstEdit, generatedEsQuery, usageTracker, onChangeUserQuery]
+    // We should not include form methods in dep arrays per react-hook-form
```

Comment:
You can have them actually. Not all the methods but recommended way is to use destructuring.

```typescript
// don't do
const methods = useForm()
useEffect(() => {
...
}, [methods]);

// do 
const { reset } = useForm();

useEffect(() => {
...
}, [reset]);
```

https://react-hook-form.com/docs/useform
<img width="931" alt="Screenshot 2025-06-17 at 23 29 06" src="https://github.com/user-attachments/assets/c8025c73-9501-4f67-bfa9-b8611bd500b0" />